File: palindrome_number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if (x == 0) {
            return true;
        }
        if (x < 0) {
            return false;
        }
        int numToReverse = x;
        long reversed = 0;
        while (numToReverse != 0) {
            int digit = numToReverse % 10;
            reversed = reversed * 10 + digit;
            numToReverse = numToReverse / 10;
        }
        if(reversed == x) {
            return true;
        } else {
            return false;
        }
    }
};

File: palindrome_number.java
import java.util.Scanner;

public class Solution {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        int x = input.nextInt();
        System.out.println(isPalindrome(x));
    }

    public static boolean isPalindrome(int x) {
        if (x < 0) {
            return false; // Negative integers are not palindromes
        }
        int original = x;
        int reversed = 0;
        while (x != 0) {
            reversed = reversed * 10 + x % 10;
            x /= 10;
        }
        return original == reversed;
    }
}
